AutoGrade/models.py

```diff
@@ -167,9 +188,13 @@ def moss_submit(self):
             logging.debug("MOSS: No submissions available for generating Moss report")
             return False
 
-        m = mosspy.Moss(settings.MOSS_USERID, "python")
+        if not self.language:
+            raise EnvironmentError("Language not assigned for Assignment")
+
+        assignment_extension = self.assignment_file.url.split(".")[-1] # get extension of assignment files
+        m = mosspy.Moss(settings.MOSS_USERID, assignment.language.moss_name)
```

Comment:
This line need to have backward compatibility to use Python by default. In case when assignment are already added in database they will have no language assignment, this is an problem.